### PR TITLE
Shells rabbit

### PR DESCRIPTION
**Pull request by stristr**
OK to merge.

Short review.

Summary of changes:

- Small tweak/fix for #862—gives `prettyValue` a normalized structure so that we can always use it like an object.
 - Some minor rabbit/hat work for shells-ui—adds an `IF_ROOT_CHILD` check for property display/row hydration and applies it to layout properties. Also hides some sketchy 2.5D (z-related) properties.

Regressions to look for:

- Please test carefully, but this should simply not populate rows for properties we don't want to display.

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality

### CODE REVIEW

**Comment by stristr** [comment]
@roperzh, rebased to `master` and also realized I had to revert your last two commits in the original `shells-ui` merge since #867 makes it no longer needed.

File: packages/haiku-serialization/src/bll/Row.js [hunk 1]
```diff
@@ -605,7 +605,9 @@ class Row extends BaseModel {
     let depth = 0
     let parent = this.parent
     while (parent) {
-      depth += 1
+      if (parent.element.hasAddressableProperties) {
+        depth += 1
+      }
```
Comment (stristr) [inline packages/haiku-serialization/src/bll/Row.js:610]: @roperzh, please holler if you think this is a terrible idea or prone to breaking something important.

The point of this commit is to turn [this](https://cl.ly/02512030746c) into [this](https://cl.ly/1de3bd4589eb). The random-seeming indentation was a function of using "pure depth" to determine display depth—arguably, the better depth to use is "depth of things I can see".

**Comment by stristr** [comment]
Correction re: 

> I had to revert your last two commits

I ended up needing it to prevent weird clearing issues in the timeline, so went with 47ec1d4de588dfa72344f2a8cfe63fedb3c61a39 instead.

File: packages/haiku-serialization/src/bll/Row.js (hunk 1 above)
Comment (roperzh) [reply packages/haiku-serialization/src/bll/Row.js:610]: This is way nicer, thank you.

**Review by roperzh — APPROVED** [review]
LGTM and works incredibly well.

**Comment by stristr** [comment]
Okay, going to merge this with caveat that we basically need to QA entire timeline again 🐇 🎩 !

**Pull request merged**